[UPG] Base migration upgrade for fields rename.

BT-dherreros opened the pull request.

bosd commented on odoo_module_migrate/base_migration_script.py:
```diff
@@ -208,10 +208,17 @@ def process_file(
         )
+        field_renames = renamed_fields.get("replaces")
+        # To be safe we only rename fields on files associated with the current replaces
+        if field_renames:
```
This creates an UnboundLocalError:
```
    module_migration.run()
    ~~~~~~~~~~~~~~~~~~~~^^
  File "/home/data/git/OCA/odoo-module-migrator/odoo_module_migrate/module_migration.py", line 34, in run
    migration_script.run(
    ~~~~~~~~~~~~~~~~~~~~^
        self._module_path,
        ^^^^^^^^^^^^^^^^^^
    ...<4 lines>...
        self._migration._commit_enabled,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/home/data/git/OCA/odoo-module-migrator/odoo_module_migrate/base_migration_script.py", line 160, in run
    self.process_file(
    ~~~~~~~~~~~~~~~~~^
        root,
        ^^^^^
    ...<4 lines>...
        commit_enabled,
        ^^^^^^^^^^^^^^^
    )
    ^
  File "/home/data/git/OCA/odoo-module-migrator/odoo_module_migrate/base_migration_script.py", line 230, in process_file
    if re.findall(pattern, new_text):
                           ^^^^^^^^
UnboundLocalError: cannot access local variable 'new_text' where it is not associated with a value

```

bosd replied:
I have merged this pr in my own fork. Then used  it to migrate some modules from the brand repo.

bosd reviewed (approved):
Code review of the fix. No func test yet.

BT-rmartin commented:
@CLaurelB @legalsylvain @lef-adhoc Could you please review this PR?

BT-dherreros commented:
@CLaurelB @legalsylvain @lef-adhoc I've updated this code with master, cleaned it slightly and also introduced data to be tested so the code coverage is improved

BT-rmartin reviewed (approved):
Excellent job!

BT-rmartin commented:
@gurneyalex @dreispt @simahawk Could you please review and merge it?

BT-rmartin commented:
@sebalix Cna you please re-review?
